Source/QED/BreitWheelerEngineWrapper.cpp

```diff
@@ -1,16 +1,205 @@
+BreitWheelerEngine::build_optical_depth_functor ()
 {
     return BreitWheelerGetOpticalDepth();
 }
 
+BreitWheelerEvolveOpticalDepth
+BreitWheelerEngine::build_evolve_functor ()
+{
+    AMREX_ALWAYS_ASSERT(lookup_tables_initialized);
+
+    return BreitWheelerEvolveOpticalDepth(innards.ctrl,
+        innards.TTfunc_coords.size(),
+        innards.TTfunc_coords.dataPtr(),
+        innards.TTfunc_data.dataPtr() );
+}
+
+BreitWheelerGeneratePairs
+BreitWheelerEngine::build_pair_functor ()
+{
+    AMREX_ALWAYS_ASSERT(lookup_tables_initialized);
+
+    return BreitWheelerGeneratePairs(innards);
+}
+
+bool BreitWheelerEngine::are_lookup_tables_initialized () const
+{
+    return lookup_tables_initialized;
+}
+
+bool
+BreitWheelerEngine::init_lookup_tables_from_raw_data (
+    const Vector<char>& raw_data)
+{
+    const char* p_data = raw_data.data();
+    const char* const p_last = &raw_data.back();
+    bool is_ok;
+
+    //Header (control parameters)
+    tie(is_ok, innards.ctrl.chi_phot_min, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_min)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tdndt_min, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tdndt_min)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tdndt_max, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tdndt_max)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tdndt_how_many, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tdndt_how_many)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tpair_min, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tpair_min)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tpair_max, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tpair_max)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_phot_tpair_how_many, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_phot_tpair_how_many)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    tie(is_ok, innards.ctrl.chi_frac_tpair_how_many, p_data) =
+        parse_raw_data<decltype(innards.ctrl.chi_frac_tpair_how_many)>(
+            p_data, p_last);
+    if(!is_ok) return false;
+
+    //___________________________
+
+    //Data
+    Vector<Real> tndt_coords(innards.ctrl.chi_phot_tdndt_how_many);
+    Vector<Real> tndt_data(innards.ctrl.chi_phot_tdndt_how_many);
+    Vector<Real> cum_tab_coords1(innards.ctrl.chi_phot_tpair_how_many);
+    Vector<Real> cum_tab_coords2(innards.ctrl.chi_frac_tpair_how_many);
+    Vector<Real> cum_tab_data(innards.ctrl.chi_phot_tpair_how_many*
+        innards.ctrl.chi_frac_tpair_how_many);
+
+    tie(is_ok, tndt_coords, p_data) =
```

Comment:
Could you add a small comment before this line, to explain what this block is doing? Same thing for the few similar blocks below (I am just not sure of what `tndt_coords`, `tndt_data`, `cum_tab_coords1` `cum_tab_coords2` stand for), and for the QSR module. Some description could go to the `innard` files.